frontend2/src/content/ManageContent.ts

```diff
@@ -0,0 +1,42 @@
+import type {
+  DebuggingTipsPageKey,
+  TourneyPageKey,
+  QuickStartPageKey,
+  ResourcesPageKey,
+  CommonIssuesPageKey,
+  PageKey,
+} from "./ContentStruct";
+
+import * as BC24 from "./bc24";
+import * as BC23 from "./bc23";
+import * as BC22 from "./bc22";
+import * as BC24TEST from "./bc24test";
+
+const bcVersions: Record<
+  string,
+  typeof BC24 | typeof BC23 | typeof BC22 | typeof BC24TEST
+> = {
+  bc24: BC24,
+  bc23: BC23,
+  bc22: BC22,
+  bc24test: BC24TEST,
+};
+
+function createTextRecord<K extends PageKey>(
```

Comment:
curious... why extends pagekey and not just pagekey here? if this is intentional, then just disregard tho

Reply:
it is intentional, It comes from the reason that I wanted for every text like quickStartText, resourcesText to have type record<QuickStartPageKey, string>, record<ResourcesPageKey, string> .. instead of the general PageKey Type